glotaran/builtin/megacomplexes/damped_oscillation/test/test_doas_model.py

```diff
@@ -6,43 +6,23 @@
 from glotaran.builtin.megacomplexes.damped_oscillation import DampedOscillationMegacomplex
 from glotaran.builtin.megacomplexes.decay import DecayMegacomplex
 from glotaran.builtin.megacomplexes.spectral import SpectralMegacomplex
-from glotaran.model import Megacomplex
 from glotaran.model import Model
 from glotaran.optimization.optimize import optimize
-from glotaran.parameter import ParameterGroup
+from glotaran.parameter import Parameters
 from glotaran.project import Scheme
 from glotaran.simulation import simulate
 
-
-class DampedOscillationsModel(Model):
-    @classmethod
-    def from_dict(
-        cls,
-        model_dict,
-        *,
-        megacomplex_types: dict[str, type[Megacomplex]] | None = None,
-        default_megacomplex_type: str | None = None,
-    ):
-        defaults: dict[str, type[Megacomplex]] = {
-            "damped_oscillation": DampedOscillationMegacomplex,
-            "decay": DecayMegacomplex,
-            "spectral": SpectralMegacomplex,
-        }
-        if megacomplex_types is not None:
-            defaults.update(megacomplex_types)
-        return super().from_dict(
-            model_dict,
-            megacomplex_types=defaults,
-            default_megacomplex_type=default_megacomplex_type,
-        )
+DampedOscillationsModel = Model.create_class_from_megacomplexes(
+    [DampedOscillationMegacomplex, DecayMegacomplex, SpectralMegacomplex]
+)
 
 
 class OneOscillation:
-    sim_model = DampedOscillationsModel.from_dict(
-        {
+    sim_model = DampedOscillationsModel(
+        **{
             "megacomplex": {
                 "m1": {
-                    "type": "damped_oscillation",
+                    "type": "damped-oscillation",
```

Comment:
The example in pyglotaran-examples is also using the `-` syntax, so this change makes it more consistent.